website/notifications/utils.py

```diff
@@ -236,9 +236,11 @@ def format_data(user, node_ids):
                              if getattr(subscription, 'event_name') in node_sub_available]
             for subscription in subscriptions:
                 index = node_sub_available.index(getattr(subscription, 'event_name'))
-                children.append(serialize_event(user, subscription=subscription,
+                if not (not node.mailing_enabled and subscription.event_name == 'mailing_list_events'):
```

Comment:
It makes the notification serialization omit MLE-type subscriptions iff the node's mailing list is disabled, so that the option to change them is not displayed in the treebeard on either settings page.

`not node.mailing_enabled` can be thought of (perhaps more helpfully) as `node.mailing_disabled`, so this condition is, in English, `If this event is not a mailing list event on a node with mailing lists disabled: serialize it`, which I'll add as a comment.